x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -79,105 +102,329 @@ export class AlertsClient {
     id,
+    query,
     index,
-  }: GetAlertParams): Promise<(AlertType & { _version: string | undefined }) | null | undefined> {
+    operation,
+    lastSortIds = [],
+  }: SingleSearchAfterAndAudit) {
     try {
       const alertSpaceId = this.spaceId;
       if (alertSpaceId == null) {
-        this.logger.error('Failed to acquire spaceId from authorization client');
-        return;
+        const errorMessage = 'Failed to acquire spaceId from authorization client';
+        this.logger.error(`fetchAlertAndAudit threw an error: ${errorMessage}`);
+        throw Boom.failedDependency(`fetchAlertAndAudit threw an error: ${errorMessage}`);
       }
+
+      const config = getEsQueryConfig();
+
+      let queryBody = {
+        query: await this.buildEsQueryWithAuthz(query, id, alertSpaceId, operation, config),
+        sort: [
+          {
+            '@timestamp': {
+              order: 'asc',
+              unmapped_type: 'date',
+            },
+          },
+        ],
+      };
+
+      if (lastSortIds.length > 0) {
+        queryBody = {
+          ...queryBody,
+          // @ts-expect-error
+          search_after: lastSortIds,
+        };
+      }
+
       const result = await this.esClient.search<ParsedTechnicalFields>({
-        // Context: Originally thought of always just searching `.alerts-*` but that could
-        // result in a big performance hit. If the client already knows which index the alert
-        // belongs to, passing in the index will speed things up
         index: index ?? '.alerts-*',
         ignore_unavailable: true,
-        body: {
-          query: {
-            bool: {
-              filter: [{ term: { _id: id } }, { term: { [SPACE_IDS]: alertSpaceId } }],
-            },
-          },
-        },
+        // @ts-expect-error
+        body: queryBody,
         seq_no_primary_term: true,
       });
 
-      if (result == null || result.body == null || result.body.hits.hits.length === 0) {
-        return;
+      if (!result?.body.hits.hits.every((hit) => isValidAlert(hit._source))) {
+        const errorMessage = `Invalid alert found with id of "${id}" or with query "${query}" and operation ${operation}`;
+        this.logger.error(errorMessage);
+        throw Boom.badData(errorMessage);
       }
 
-      if (!isValidAlert(result.body.hits.hits[0]._source)) {
-        const errorMessage = `Unable to retrieve alert details for alert with id of "${id}".`;
-        this.logger.debug(errorMessage);
-        throw new Error(errorMessage);
-      }
+      await this.ensureAllAuthorized(result.body.hits.hits, operation);
 
-      return {
-        ...result.body.hits.hits[0]._source,
-        _version: encodeHitVersion(result.body.hits.hits[0]),
-      };
+      result?.body.hits.hits.map((item) =>
+        this.auditLogger?.log(
+          alertAuditEvent({
+            action: operationAlertAuditActionMap[operation],
+            id: item._id,
+            outcome: 'unknown',
```

Comment:
As mentioned in my other comment on the unit test: the outcome should only be 'unknown' if the action is an update.
If `outcome` is omitted and `error` is not present, the audit event is written as a a successful event (`outcome: 'success'`)
```suggestion
            ...(operation === WriteOperations.Update ? { outcome: 'unknown' } : {}),
```

Reply:
Fixed here: 36838ab2e10ba9b2640665345758373075e0fa44